src/pretix/base/forms/questions.py

```diff
@@ -288,11 +288,18 @@ def __init__(self, *args, **kwargs):
                     initial=initial.options.all() if initial else None,
                 )
             elif q.type == Question.TYPE_FILE:
+                # Browsers can't recognize that the server already has a file uploaded
+                # Don't mark this input as being required if we already have an answer
+                # (this needs to be done via the attrs, otherwise we wouldn't get the "required" star on the field label)
+                if required and not initial:
+                    widget = UploadedFileWidget(position=pos, event=event, answer=initial, attrs={'required': None})
```

Comment:
Derp, seems I refactored it the wrong way around, but you get the idea :-)

The other place I could see it added would be in the `UploadedFileWidget`. Which spot do you think would be best?

I should be able to recheck this tonight and to put it in `UploadedFileWidget` if you think it should go there rather than here.

Reply:
The widget would probably be even better, yes.